etcdmgmt/etcd-mgmt_test.go

```diff
@@ -33,33 +62,33 @@ func TestWriteETCDPidFile(t *testing.T) {
 	defer heketitests.Patch(&etcdPidFile, "/a/b/c/d/etcd.pid").Restore()
 	tests.Assert(t, writeETCDPidFile(cmd.Process.Pid) != nil)
 	cmd.Process.Kill()
+	cmd.Process.Wait()
 }
 
 func TestIsETCDStartNeeded(t *testing.T) {
 	// check once etcd process is running isETCDStartNeeded returns false
 	os.Remove(etcdPidFile)
-	cmd := exec.Command("etcd")
+	cmd := formETCDCommand()
 	err := cmd.Start()
 	tests.Assert(t, err == nil)
 	err = writeETCDPidFile(cmd.Process.Pid)
 	tests.Assert(t, err == nil)
+	time.Sleep(15 * time.Second)
```

Comment:
We'd need to give some time window to etcd to get it settled. `checkHealth` function exactly does the same and the max time out is 15 secs.